src/service/action-impl.js

```diff
@@ -733,23 +735,31 @@ export class ActionService {
   }
 }
 
-
 /**
  * Returns `true` if the given action invocation is whitelisted in the given
- * whitelist.
+ * whitelist. Default actions' alias, 'activate', are automatically
+ * whitelisted if their corresponding registered alias is whitelisted.
  * @param {!ActionInvocation} invocation
  * @param {!Array<{tagOrTarget: string, method: string}>} whitelist
  * @return {boolean}
  * @private
  */
 function isActionWhitelisted_(invocation, whitelist) {
+  const tagOrTargetRegex = new RegExp('^' + invocation.tagOrTarget + '$', 'i');
+  let invocationMethodRegex = new RegExp('^' + invocation.method + '$', 'i');
+  const component = invocation.node.implementation_;
+  if (invocation.method == DEFAULT_METHOD && component
+      && component.defaultActionAlias_) {
+    invocationMethodRegex = new RegExp(
+        '^' + component.defaultActionAlias_ + '$', 'i');
+  }
```

Comment:
This looks messy, not to mention the inspection of private variables externally.

Reply:
right. I added the necessary API calls to expose the private property. Thanks